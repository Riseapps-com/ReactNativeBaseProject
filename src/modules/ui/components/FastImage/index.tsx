import React from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import NativeFastImage, { FastImageProps as NativeFastImageProps } from 'react-native-fast-image';

export type FastImageProps = NativeFastImageProps & {
  style: StyleProp<ViewStyle>;
};

const FastImage: React.FC<FastImageProps> = props => {
  return <NativeFastImage {...props} style={props.style} />;
};

export default FastImage;
