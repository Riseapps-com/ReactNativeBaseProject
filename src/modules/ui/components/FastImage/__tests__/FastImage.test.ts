import { images } from '~assets';
import { renderComponent } from '~modules/tests';

import FastImage, { FastImageProps } from '../index';

const defaultProps: FastImageProps = {
  source: images.americas,
};

const renderFastImage = () => renderComponent(FastImage, defaultProps);

describe('ui', () => {
  describe('<FastImage />', () => {
    it('renders <FastImage />', () => {
      const fastImage = renderFastImage();

      expect(fastImage).toMatchSnapshot();
    });
  });
});
