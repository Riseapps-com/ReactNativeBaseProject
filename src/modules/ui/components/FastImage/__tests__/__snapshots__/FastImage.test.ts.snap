// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <FastImage /> renders <FastImage /> 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../assets/images/america.png",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
